src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchTrialDAO.scala

```diff
@@ -273,23 +274,75 @@ class ElasticSearchTrialDAO(client: TransportClient, indexName: String, refreshM
     )
   }
 
+  private def handleScrollResponse(scrollResponse: SearchResponse): (String, List[TrialProject]) = {
+    val scrollId: String = scrollResponse.getScrollId
+
+    val trialProjects: List[TrialProject] = if (scrollResponse.getHits.totalHits == 0) {
+      List.empty[TrialProject]
+    } else {
+      scrollResponse.getHits.getHits.toList map ( _.getSourceAsString.parseJson.convertTo[TrialProject] )
+    }
+
+    (scrollId, trialProjects)
+  }
+
+  private def startScroll: (String, List[TrialProject]) = {
+    val startScrollRequest = client
+      .prepareSearch(indexName)
+      .setQuery(Ready)
+      .setScroll(TimeValue.timeValueMinutes(1))
+      .addSort("_doc", SortOrder.ASC)
+      .setSize(250) // tweak pagesize here
+
+    handleScrollResponse(
+      executeESRequest[SearchRequest, SearchResponse, SearchRequestBuilder](startScrollRequest))
+  }
+
+  private def continueScroll(scrollId: String): (String, List[TrialProject]) = {
+    val continueScrollRequest = client
+      .prepareSearchScroll(scrollId)
+      .setScroll(TimeValue.timeValueMinutes(1))
+
+    handleScrollResponse(
+      executeESRequest[SearchScrollRequest, SearchResponse, SearchScrollRequestBuilder](continueScrollRequest))
+  }
+
+  private def clearScroll(scrollId: String): Future[Any] = {
+    // we want to clear the scroll to release resources on the ES server.
+    // however, ES will automatically clean up scrolls after their timeout.
+    // so, we make this a fire-and-forget request, and don't care whether or not it succeeds or fails.
+    val clearScrollRequest = client
+      .prepareClearScroll()
+      .addScrollId(scrollId)
+    Future(executeESRequest[ClearScrollRequest, ClearScrollResponse, ClearScrollRequestBuilder](clearScrollRequest))
+  }
+
   /**
     * Returns a list of all project records.
     * @return list of all project records.
     */
   override def projectReport: Seq[TrialProject] = {
-    val reportProjectRequest = client
-      .prepareSearch(indexName)
-      .setQuery(Ready)
-      .addSort("name.keyword", SortOrder.ASC)
-      .setSize(5000) // we should loop and make paginated requests; this will have to suffice for now
 
-    val reportProjectResponse = executeESRequest[SearchRequest, SearchResponse, SearchRequestBuilder](reportProjectRequest)
+    def nextScroll(scrollId: Option[String], accum: List[TrialProject]): (String, List[TrialProject]) = {
+      // is this the initial query?
+      val (newScrollId, batch) = scrollId match {
+        case None => startScroll
+        case Some(s) => continueScroll(s)
+      }
 
-    if (reportProjectResponse.getHits.totalHits == 0)
-      Seq.empty[TrialProject]
-    else
-      reportProjectResponse.getHits.getHits.toSeq map ( _.getSourceAsString.parseJson.convertTo[TrialProject] )
+      // if the query returned results, add them to the accumulator and make another query
+      if (batch.nonEmpty) {
+        nextScroll(Some(newScrollId), accum ++ batch)
```

Comment:
Scala `List` is optimized for prepending; `++` doesn't have great performance. Though I suspect it's not too much of an issue in this cause, where each batch is a webservice call.

Cats [Chain](https://typelevel.org/cats/datatypes/chain.html) is a nice data structure which has the semantics of `List` but allows constant time prepend and append.